Add two documents for the internals

LiangrunDa opened the pull request.

ept commented:
Hi @LiangrunDa, thanks for this contribution. I have a few editorial requests to make the new documents fit better within the website:

* Instead of using your own name in the examples, could you please use some generic name so that it's more neutral.
* The first few paragraphs of `runtime.md` overlap with content already covered in other pages of the documentation site. Please could you edit it so that this page only contains information not already covered elsewhere.
* Footnote 1 of `runtime.md` references the documentation site itself in a footnote, which doesn't really make sense. If you want to refer to another page, you can just use a hyperlink.
* In `storage.md` it might make sense to include a reference to the [detailed binary format spec](https://automerge.org/docs/how-it-works/binary-format/) at the beginning, and also to make it a regular link rather than a footnote.

LiangrunDa commented:
Hi @ept , I've fixed the issues you've mentioned. 

- changed them to `Alice`, `Bob` and `Carol`.
- removed the `Overview` section
- removed the footnote
- included a introduction of binary spec in the first section of `storage.md`

ept commented:
Thank you @LiangrunDa!
